### ISSUE TITLE
i64 snowflake ID may going to negative value

### ISSUE THREAD

**Issue by ikai104** [MrGunflame/snowflaked-rs#1]
The document mention [i64 uses 41 bits for the timestamp](https://docs.rs/snowflaked/1.0.0/snowflaked/index.html#snowflake-structure). So I expecting it should wrap when 41 bits timestamp overflowed, but the following code panic.

```rust
    #[test]
    fn should_not_get_neg_value_when_using_i64() {
        use std::time::{Duration, UNIX_EPOCH};

        let epoch = UNIX_EPOCH - Duration::from_secs(60 * 60 * 24 * 365 * 20);
        let generator: Generator = Generator::builder().epoch(epoch).build();

        assert!(generator.generate::<i64>() >= 0);
    }
```

See also: https://github.com/twitter-archive/snowflake/tree/b3f6a3c6ca8e1b6847baa6ff42bf72201e2c2231#solution

**Comment by MrGunflame** [comment]
Thanks for reporting this, fixed in 854ac25b19439df9bb1fbd4e1885fff5c7a5aad5.

**Comment by MrGunflame** [comment]
The fix is released in `1.0.1`.
